source/utils/date.cpp

```diff
@@ -132,7 +117,7 @@ int date::weekday() const
     tm.tm_year = year - 1900;
     std::time_t time = std::mktime(&tm);
```

Comment:
Should we verify if `mktime` succeeded? If not, we could e.g. return -1 too to indicate failure.

Reply:
I'm not so sure this is a good idea, and I'm honestly surprised that [the documentation on cppreference](https://en.cppreference.com/w/cpp/chrono/c/mktime) states that -1 is a sign of failure. Under MSVC, this is indeed the case, because MSVC cannot use dates before 1970-1-1, and -1 would be 1969-12-31 23:59:59. However, for other platforms, it is often the case that dates at least back to 1900-1-1 are supported, and -1 is simply one second before 1970-1-1.

In other words, I would only trust -1 to be a sign of failure under MSVC. However, I would be interested if the C specification gives any more details on this.

Reply:
Apparently, [the C23 specification](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3088.pdf) still states that `If the calendar time cannot be represented, the function returns the value (time_t) (-1)`. I'm seriously confused by this and seriously doubt that this is actually the case on any other compilers than MSVC. Unless, of course, this still happens although -1 could be a valid value, in which case it's not possible on such compilers to differentiate between the actual second -1 being 1969-12-31 23:59:59 and an actual error case.

Reply:
Maybe the solution provided in [this doc](https://pubs.opengroup.org/onlinepubs/9799919799/functions/mktime.html) is a good solution:

> Since (time_t)-1 is a valid return value for a successful call to mktime(), an application wishing to check for error situations should set tm_wday to a value less than 0 or greater than 6 before calling mktime(). On return, if tm_wday has not changed an error has occurred.

Reply:
That solution is a bit cursed, but I like it better than no error handling at all. Thanks! :smile: